### PR TITLE
Updated the equalizer plugin to use sample exact controls

### PR DESCRIPTION
**Pull request by curlymorphic**
Updated the equalizer plugin to use sample exact controls for the frequency res and bandwidth controls.
For this i had to do some major refactoring within the processAudioBuffer()
function. Originally each filter processed the buffer in a separate loop, This has been improved and each filter is now processed per frame.

### CODE REVIEW

**Comment by Sti2nd** [comment]
:+1:

**Comment by tresf** [comment]
@curlymorphic, do you need review on this prior to merging?  Since EQ is something you created, I'm inclined to merge any PRs you send against it, but I also know you and Vesa were discussing this quite a bit, so I don't want to prematurely merge.

**Comment by curlymorphic** [comment]
> if you're hoping for feedback first.

I fell it may be better if @diizy give's this the once over, I have followed the guidelines for including sample exactness but not had any code doing so "passed" yet.

File: plugins/Eq/EqFilter.h [hunk 1]
```diff
@@ -136,6 +136,12 @@ class EqFilter : public StereoBiQuad
 		}
 	}
 
+	virtual void tick( sampleFrame *frame )
```
Comment (diizy) [inline plugins/Eq/EqFilter.h]: What is the purpose of this function? Just seems like adding extra overhead... would probably be better to use inline functions for things such as this

**Comment by curlymorphic** [comment]
Removed the tick() function, and replaced the invoking code with
direct calls to update().

fixed a bug, where the low shelf bandwidth was getting the incorrect data.

**Pull request merged**